### ISSUE TITLE
[Common Lisp] Implement new Concept Exercise: sets

### ISSUE THREAD

**Issue by TheLostLambda** [exercism/v3#1766]
This issue describes how to implement the `sets` concept exercise for the Common Lisp track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Learning objectives

- Know that the CL standard doesn't have a "set" datatype, but lists are used instead
- Know how to add elements to a set using `adjoin` and `pushnew`
- Understand the `:key`, `:test`, and `:test-not` keyword arguments to the set functions
- Be able to test if an element is in a set or not using `member` (or `member-if`)
- Understand how to use the `subsetp` predicate
- Be able to get the size of a set using `length` or `list-length`
- Be aware of the `intersection`, `set-difference`, `union` and `set-exclusive-or` operations
- Know that the "n"-variants (like `nunion`) mutate their arguments

## Out of scope

- Performance implications of using cons-cells for sets
- Any other list or sequence functions (not usually needed by sets)
- Hash tables or other ways to implement a set in Common Lisp

## Concepts

- `sets`

## Prerequisites

- `basic-lists`
- `higher-order-functions`
- `cons`
- `sameness`
- `function-definition`

### Hints

- [Tutorials Point](https://www.tutorialspoint.com/lisp/lisp_set.htm)

### After

Nothing yet

## Representer

Nothing yet

## Analyzer

Nothing yet

## Implementing

This is one of the easier concept exercises to implement, as it only involves making the programmer aware of the correct function names for well-known set operations. Any simple exercise with sets will do.

The only pitfall to look out for will be sets of values that aren't `eql` (like lists, strings, etc) as the student will need to manually specify a `:test` function.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.
